### DISCUSSION TITLE
RadioSet with Horizontal RadioButtons?

### DISCUSSION THREAD

**Discussion by clvnkhr** [Textualize/textual#5240] [Q&A] [answered]
I copied the source of RadioSet, replaced ScrollableContainer with Horizontal and it seems to work:
<img width="235" alt="Screenshot 2024-11-15 at 19 39 15" src="https://github.com/user-attachments/assets/555d02b9-67e4-48e3-b420-30e38ea562bf">
but Im guessing theres a better way?

[edit sorry VerticalScroll]

**Comment by TomJGooding** [comment, marked as answer]
I think `RadioSet` inherits from `VerticalScroll` in the latest version? This is maybe a bit hacky, but saves having to create your own widget:

```python
from textual.app import App, ComposeResult
from textual.widgets import RadioSet


class ExampleApp(App):
    CSS = """
    RadioSet {
        layout: horizontal;
        overflow-y: hidden;
        overflow-x: auto;
    }
    """

    def compose(self) -> ComposeResult:
        yield RadioSet("foo", "bar", "baz")


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Reply by clvnkhr** [reply]
Ah yes sorry I meant VerticalScroll 😅 and that works great thanks a bunch!!
